doc/sphinx/src/fortran.rst

```diff
@@ -0,0 +1,693 @@
+   end function get_sg_BulkModulusFromDensityInternalEnergy_f
+
+Calculates bulk modulus from density and specific internal energy.
+
+- ``matindex``: Material index (1-based in Fortran)
+- ``eos``: EOS array
+- ``rhos``: Input density array
+- ``sies``: Input specific internal energy array
+- ``bmods``: Output bulk modulus array
+- ``len``: Number of elements to process
+- ``stride``: Optional stride for processing elements
+- ``lambda_data``: Optional array for additional parameters
+
+.. code-block:: fortran
+
+   integer function get_sg_eos_f(nmat, ncell, cell_dim,&
+                                option,&
+                                eos_offsets,&
+                                eos,&
+                                offsets,&
+                                press, pmax, vol, spvol, sie, temp, bmod,&
+                                dpde, cv,&
+                                frac_mass, frac_vol, frac_sie,&
+                                frac_bmod, frac_dpde, frac_cv,&
+                                mass_frac_cutoff) &
+     result(err)
+     integer(kind=c_int), intent(in) :: nmat
+     integer(kind=c_int), intent(in) :: ncell
+     integer(kind=c_int), intent(in) :: cell_dim
+     integer(kind=c_int), intent(in) :: option
+     integer(kind=c_int), dimension(:), target, intent(in) :: eos_offsets
+     type(sg_eos_ary_t), intent(in)  :: eos
+     integer(kind=c_int), dimension(:), target, intent(in) :: offsets
+     real(kind=8), dimension(:),   target, intent(in)    :: press
+     real(kind=8), dimension(:),   target, intent(in)    :: pmax
+     real(kind=8), dimension(:),   target, intent(in)    :: vol
+     real(kind=8), dimension(:),   target, intent(in)    :: spvol
+     real(kind=8), dimension(:),   target, intent(in)    :: sie
+     real(kind=8), dimension(:),   target, intent(in)    :: temp
+     real(kind=8), dimension(:),   target, intent(in)    :: bmod
+     real(kind=8), dimension(:),   target, intent(in)    :: dpde
+     real(kind=8), dimension(:),   target, intent(in)    :: cv
+     real(kind=8), dimension(:,:), target, intent(in) :: frac_mass
+     real(kind=8), dimension(:,:), target, intent(inout) :: frac_vol
+     real(kind=8), dimension(:,:), target, intent(inout) :: frac_sie
+     ! optionals
+     real(kind=8), dimension(:,:), target, optional, intent(inout) :: frac_bmod
+     real(kind=8), dimension(:,:), target, optional, intent(inout) :: frac_dpde
+     real(kind=8), dimension(:,:), target, optional, intent(inout) :: frac_cv
+     real(kind=8),                         optional, intent(in)    :: mass_frac_cutoff
+   end function get_sg_eos_f
+
+Main EOS evaluation function for mixed cells. This function is used to calculate various thermodynamic quantities for multiple materials in mixed cells.
+
+- ``nmat``: Number of materials
+- ``ncell``: Number of cells
+- ``cell_dim``: Cell dimension
+- ``option``: Option flag for calculation
```

Comment:
In the fullness of time, this function may need significantly more discussion. But I wouldn't add that now.

I also wanted to let you know know this particular function may change, as it was originally written with one downstream code in mind. So we're exploring what this should look like in general. (And feedback is welcome.)